mbed_tools_ci/generate_docs.py

```diff
@@ -11,17 +11,13 @@
 
 
 def generate_docs(output_directory: str) -> int:
-    """Triggers building the documentation.
-
-    Module to document and the output destination path
-    can be set in the mbed_tools_ci.utils.definitions config file.
-    """
+    """Triggers building the documentation."""
     command_list = [
-        "pdoc", "--html",
-        f"{configuration.get_value(ConfigurationVariable.MODULE_TO_DOCUMENT)}",
-        "--output-dir",
-        f'{output_directory}', "--force", "--config",
-        "show_type_annotations=True"
+        "pdoc",
+        "--html", f"{configuration.get_value(ConfigurationVariable.MODULE_TO_DOCUMENT)}",
+        "--output-dir", f'{output_directory}',
+        "--force",
```

Comment:
It overwrites existing documents in that directory. I am tempted to keep it so that when generating docs locally you don't always need to manually delete the contents of the local docs folder. It doesn't however remove any deleted pages so I have to do the remove and move dance lower down for syncing to prod.

Reply:
D'oh! I did not see this option in the cli yesterday. Mea culpa. 🤦‍♂ 

I wonder if it wouldn't be cleaner if both generating docs for release and `generate-docs` didn't follow exact same path. Is there ever a case where you'd like to keep old docs? I think seeing stale files locally might get confusing when debugging.

How about making this script always clean output directory (if exists) with a `logging.info('Docs found in {output_dir}, removing old documentation first')` or similar?

Reply:
Yeah I think you are right. I will get the process right in `tag_and_release` and then add something similar to `generate_docs`.

Reply:
I will do that in a separate PR if thats ok so I can test it without bloating this one.